pkg/controller/templateupdaterequest/templateupdaterequest_controller.go

```diff
@@ -0,0 +1,320 @@
+		// if err := r.client.Status Delete(context.TODO(), tur); err != nil {
+		// 	logger.Error(err, "Unable to delete the TemplateUpdateRequest")
+		// 	return reconcile.Result{}, err
+		// }
+		return reconcile.Result{Requeue: true, RequeueAfter: DeletionTimeout}, r.updateStatusConditions(logger, tur, toBeComplete())
+	}
+	// otherwise, we need to wait
+	logger.Info("MasterUserRecord still being updated...")
+	return reconcile.Result{Requeue: true, RequeueAfter: ProgressDelay}, nil
+}
+
+func (r ReconcileTemplateUpdateRequest) updateTemplateRefs(logger logr.Logger, tur toolchainv1alpha1.TemplateUpdateRequest, mur *toolchainv1alpha1.MasterUserRecord) error {
+	// update MasterUserRecord' accounts whose tier matches the TemplateUpdateRequest
+	for i, ua := range mur.Spec.UserAccounts {
+		if ua.Spec.NSTemplateSet.TierName == tur.Spec.TierName {
+			logger.Info("updating templaterefs", "tier", tur.Spec.TierName, "target_cluster", ua.TargetCluster)
+			// reset the new templateRefs, only retain those with a custom template in use
+			namespaces := make(map[string]*toolchainv1alpha1.NSTemplateSetNamespace, len(ua.Spec.NSTemplateSet.Namespaces))
+			for _, ns := range ua.Spec.NSTemplateSet.Namespaces {
+				if ns.Template != "" {
+					t := namespaceType(ns.TemplateRef)
+					namespaces[t] = &ns
+				}
+			}
+			// now, add the new templateRefs, uneless there's a custom template in use (for the same ns type)
+			for _, ns := range tur.Spec.Namespaces {
+				t := namespaceType(ns.TemplateRef)
+				if _, found := namespaces[t]; found {
+					// don't override the custom template
+					namespaces[t].TemplateRef = ns.TemplateRef
+				} else {
+					namespaces[t] = &toolchainv1alpha1.NSTemplateSetNamespace{
+						TemplateRef: ns.TemplateRef,
+					}
+				}
+			}
+			// finally, set the new namespace templates in the user account
+			ua.Spec.NSTemplateSet.Namespaces = []toolchainv1alpha1.NSTemplateSetNamespace{}
+			for _, ns := range namespaces {
+				ua.Spec.NSTemplateSet.Namespaces = append(ua.Spec.NSTemplateSet.Namespaces, *ns)
+			}
+			// now, let's take care about the cluster resources
+			if ua.Spec.NSTemplateSet.ClusterResources != nil && ua.Spec.NSTemplateSet.ClusterResources.Template != "" {
+				// retain the custom template
+				if tur.Spec.ClusterResources != nil {
+					ua.Spec.NSTemplateSet.ClusterResources.TemplateRef = tur.Spec.ClusterResources.TemplateRef
+				} else {
+					ua.Spec.NSTemplateSet.ClusterResources.TemplateRef = ""
+				}
+			} else if tur.Spec.ClusterResources != nil {
+				ua.Spec.NSTemplateSet.ClusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
+					TemplateRef: tur.Spec.ClusterResources.TemplateRef,
+				}
+			} else {
+				ua.Spec.NSTemplateSet.ClusterResources = nil
+			}
+			mur.Spec.UserAccounts[i] = ua
+		}
+		// also, update the tier template hash label
+		hash, err := nstemplatetier.ComputeHashForTemplateUpdateRequest(tur)
```

Comment:
IIRC, that's because the `NSTemplateSet` object is not available in this func (only the `NSTemplateSetSpec`), so I added this func which uses the same values

Reply:
That's correct, but the function `ComputeHashForNSTemplateSetSpec` expects the `NSTemplateSetSpec` type
https://github.com/codeready-toolchain/host-operator/blob/bd96147b5eaa2b5313ac8dd87ac5e9bcb84692ca/pkg/controller/nstemplatetier/label_selector.go#L77-L87

Reply:
done in d4b8267